src/model/store.coffee

```diff
@@ -59,6 +59,19 @@ class Request extends Varying
       # TODO: This should generate an object too. How does that happen?
       response
 
+  # Returns a promise for the deserialized request result. If the request
+  # fails, the promise is rejected with the request state.
+  asPromise: ->
+    new Promise((resolve, reject) =>
```

Comment:
I think generally having bridges to Rx/Promise/whatever else is popular and semantically harmonious enough to merit consideration is a very good thing.

But I'd rather do it at the `Varying` level than `Request`. The reason `Request` exists as its own entity right now at all is merely to give the implementer the easy ability to attach a (point-free) behavior for attempting to fulfill a value. As a state management abstraction, `Request` differs from `Varying` in nothing but convention (ie the state-describing value wrapper being leveraged in this PR). In fact, I have thoughts about being able to make requests for remote objects from the server less like one-off pulls and more like temporary subscriptions, where the server can feed updates directly back into the ostensibly-fulfilled `Request`/`Varying` object.

So an API that looks more like `Varying.asPromise(v).succeedIf(f).failureIf(f) -> Promise` is probably what I'd be looking for. Similar for `.asObservable` for Rx.